.github/actions/deploy-windows/action.yml

```diff
@@ -138,30 +142,43 @@ runs:
         git fetch --depth 1 https://github.com/%GITHUB_REPOSITORY% %GITHUB_SHA%
         git checkout %GITHUB_SHA%
 
-        mkdir %USERPROFILE%\.m2
-        if "${{ github.event_name }}"=="pull_request" (set MAVEN_PHASE=install) else (set MAVEN_PHASE=deploy)
-        echo ^<settings^> > %USERPROFILE%\.m2\settings.xml
-        echo ^<mirrors^>^<mirror^>^<id^>google-maven-central^</id^>^<name^>GCS Maven Central mirror^</name^>^<url^>https://maven-central.storage-download.googleapis.com/maven2/^</url^>^<mirrorOf^>central^</mirrorOf^>^</mirror^>^</mirrors^> >> %USERPROFILE%\.m2\settings.xml
-        echo ^<servers^>^<server^>^<id^>sonatype-nexus-snapshots^</id^>^<username^>%CI_DEPLOY_USERNAME%^</username^>^<password^>%CI_DEPLOY_PASSWORD%^</password^>^</server^>^<server^>^<id^>sonatype-nexus-staging^</id^>^<username^>%CI_DEPLOY_USERNAME%^</username^>^<password^>%CI_DEPLOY_PASSWORD%^</password^>^</server^>^</servers^> >> %USERPROFILE%\.m2\settings.xml
-        echo ^</settings^> >> %USERPROFILE%\.m2\settings.xml
+        set MAKEJ=%NUMBER_OF_PROCESSORS%
 
-        set /a "SECONDS=%RANDOM% %% 300"
-        echo Sleeping %SECONDS% seconds to help avoid race conditions with Maven
-        timeout %SECONDS% > nul
+        if not "%CI_DEPLOY_NEED_GRADLE"=="" (
+          setlocal enabledelayedexpansion
+          if "${{ github.event_name }}"=="pull_request" (set GRADLE_TASK=build) else (set GRADLE_TASK=publish)
+          echo Executing Gradle !GRADLE_TASK! on %MAKEJ% processors
+          gradle clean !GRADLE_TASK! --info && goto :done
+          exit /b !ERRORLEVEL!
+          :done
+          echo Done executing Gradle Tasks
```

Comment:
I'm not that well versed in Batch and the build wouldn't work without the enabledelayedexpansion flag like we have for the Maven route. It would also fail to exit and fail the CI run if I didn't have this switch for the gradle call. Let me know if you know how we can work around this in a nicer way.

Reply:
Batch files don't have an `set -e` mode like Bash, so we need to add explicit checks like that, that's alright, yes. We'll probably need to execute Gradle twice like Maven to work around issues with snapshots, but I'm not sure exactly how that's going to look like yet...